cpp/src/arrow/flight/sql/odbc/tests/connection_info_test.cc

```diff
@@ -0,0 +1,1453 @@
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_IDENTIFIER_CASE) {
+  this->connect();
+
+  validate(conn, SQL_IDENTIFIER_CASE, static_cast<SQLUSMALLINT>(4));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_IDENTIFIER_QUOTE_CHAR) {
+  this->connect();
+
+  validate(conn, SQL_IDENTIFIER_QUOTE_CHAR, L"\"");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_INDEX_KEYWORDS) {
+  this->connect();
+
+  validate(conn, SQL_INDEX_KEYWORDS, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_INSERT_STATEMENT) {
+  this->connect();
+
+  validate(conn, SQL_INSERT_STATEMENT, static_cast<SQLUINTEGER>(7));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_INTEGRITY) {
+  this->connect();
+
+  validate(conn, SQL_INTEGRITY, L"N");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_KEYWORDS) {
+  this->connect();
+
+  validateNotEmptySQLWCHAR(conn, SQL_KEYWORDS, true);
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_LIKE_ESCAPE_CLAUSE) {
+  this->connect();
+
+  validate(conn, SQL_LIKE_ESCAPE_CLAUSE, L"Y");
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_NON_NULLABLE_COLUMNS) {
+  this->connect();
+
+  validate(conn, SQL_NON_NULLABLE_COLUMNS, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_OJ_CAPABILITIES) {
+  this->connect();
+
+  validate(this->conn, SQL_OJ_CAPABILITIES, static_cast<SQLUINTEGER>(7));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_ORDER_BY_COLUMNS_IN_SELECT) {
+  this->connect();
+
+  validate(conn, SQL_ORDER_BY_COLUMNS_IN_SELECT, L"Y");
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_OUTER_JOINS) {
+  this->connect();
+
+  validate(conn, SQL_OUTER_JOINS, L"N");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_PROCEDURES) {
+  this->connect();
+
+  validate(conn, SQL_PROCEDURES, L"N");
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_QUOTED_IDENTIFIER_CASE) {
+  this->connect();
+
+  validate(conn, SQL_QUOTED_IDENTIFIER_CASE, static_cast<SQLUSMALLINT>(4));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_SCHEMA_USAGE) {
+  this->connect();
+
+  validate(conn, SQL_SCHEMA_USAGE, static_cast<SQLUINTEGER>(1));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_SPECIAL_CHARACTERS) {
+  this->connect();
+
+  validate(conn, SQL_SPECIAL_CHARACTERS, L"");
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_SQL_CONFORMANCE) {
+  this->connect();
+
+  validate(conn, SQL_SQL_CONFORMANCE, static_cast<SQLUINTEGER>(1));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_SUBQUERIES) {
+  this->connect();
+
+  validate(conn, SQL_SUBQUERIES, static_cast<SQLUINTEGER>(31));
```

Comment:
Since `SUBQUERIES` is not mentioned on the [sqlite_sql_info.cc](https://github.com/Bit-Quill/arrow/blob/master/cpp/src/arrow/flight/sql/example/sqlite_sql_info.cc) page, I think the driver might be returning the default value for this.
We should check for default values that are returned by the driver